eslint.config.mjs

```diff
@@ -121,35 +206,98 @@ const config = createConfig(
       '@typescript-eslint/prefer-reduce-type-parameter': 'off',
       'no-restricted-syntax': 'off',
       'no-restricted-globals': 'off',
+
+      // TODO: These rules created more errors after the upgrade to ESLint 9.
+      // Re-enable these rules and address any lint violations.
+      '@typescript-eslint/consistent-type-exports': 'warn',
+      '@typescript-eslint/explicit-function-return-type': 'off',
+      '@typescript-eslint/no-base-to-string': 'warn',
+      '@typescript-eslint/no-duplicate-enum-values': 'warn',
+      '@typescript-eslint/no-misused-promises': 'warn',
+      '@typescript-eslint/no-unsafe-enum-comparison': 'warn',
+      '@typescript-eslint/no-unused-vars': 'warn',
+      '@typescript-eslint/only-throw-error': 'warn',
+      '@typescript-eslint/prefer-promise-reject-errors': 'warn',
+      '@typescript-eslint/prefer-readonly': 'warn',
+      '@typescript-eslint/no-shadow': 'warn',
+      'import-x/namespace': 'warn',
+      'import-x/no-named-as-default': 'warn',
+      // Overrides eslint base config which isn't following outer most pattern. Can be removed once this issue is resolved and eslint config updated
+      // issue: https://github.com/MetaMask/eslint-config/issues/403
+      'import-x/order': [
+        'error',
+        {
+          // Require newlines between different groups
+          'newlines-between': 'always',
+          // Define groups in order of "outermost inward" pattern
+          groups: [
+            // External libraries first (Node.js builtins and node_modules)
+            ['builtin', 'external'],
+            // Then parent imports (../../)
+            ['parent'],
+            // Then sibling imports (../)
+            ['sibling'],
+            // Then local imports (./) including index
+            ['index', 'internal'],
+          ],
+          // Alphabetically sort within each group
+          alphabetize: {
+            order: 'asc',
+            caseInsensitive: true,
+          },
+        },
+      ],
```

Comment:
Updating order of imports as the current eslint base configuration isn't adhering to "outermost inward" pattern and created an issue here https://github.com/MetaMask/eslint-config/issues/403